src/OAuth/OAuth1/Service/Geocaching.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace OAuth\OAuth1\Service;
+
+use OAuth\OAuth1\Signature\SignatureInterface;
+use OAuth\OAuth1\Token\StdOAuth1Token;
+use OAuth\Common\Http\Exception\TokenResponseException;
+use OAuth\Common\Http\Uri\Uri;
+use OAuth\Common\Consumer\CredentialsInterface;
+use OAuth\Common\Http\Uri\UriInterface;
+use OAuth\Common\Storage\TokenStorageInterface;
+use OAuth\Common\Http\Client\ClientInterface;
+use OAuth\Common\Exception\Exception;
+
+class Geocaching extends AbstractService
+{
+
+    const ENDPOINT_STAGING = 'http://staging.geocaching.com/OAuth/oauth.ashx';
+
+    const ENDPOINT_LIVE = 'https://www.geocaching.com/OAuth/oauth.ashx';
+
+    const ENDPOINT_LIVE_MOBILE = 'https://www.geocaching.com/oauth/mobileoauth.ashx';
+
+    const BASEAPI_STAGING = 'https://staging.api.groundspeak.com/Live/V6Beta/geocaching.svc/';
+
+    const BASEAPI_LIVE = 'https://api.groundspeak.com/LiveV6/geocaching.svc/';
+
+    public function __construct(
+        CredentialsInterface $credentials,
+        ClientInterface $httpClient,
+        TokenStorageInterface $storage,
+        SignatureInterface $signature,
+        UriInterface $baseApiUri = null
+    ) {
+        parent::__construct($credentials, $httpClient, $storage, $signature, $baseApiUri);
+        if ($baseApiUri === null) {
+            $this->setBaseApiUri();
+        }
+        $this->setEndPoint();
+    }
+
+    public function getRequestTokenEndpoint()
+    {
+        return $this->getEndPoint();
```

Comment:
Are you sure all three endpoints are supposed to be the same? It would be great if you could also provide an example (see `examples/`) directory.